gmaps/directions.py

```diff
@@ -88,17 +101,29 @@ def _handle_layer_status(self, change):
                 "No directions returned: " + change["new"])
 
 
-def _directions_options(start, end, waypoints):
+def _directions_options(start, end, waypoints, avoid_ferries=False,
+                        avoid_highways=False, avoid_tolls=False,
+                        optimize_waypoints=False):
```

Comment:
I've made the tests pass -- in the future, that test should probably be changed to test the widget directly, rather than a private method.